wasm/text/parser.go

```diff
@@ -486,6 +492,80 @@ func (p *moduleParser) parseImportFuncEnd(tok tokenType, tokenBytes []byte, _, _
 	return unexpectedToken(tok, tokenBytes)
 }
 
+// parseFuncName is the first parser inside a function field. This records the function name if present and sets the
+// next parser to parseFunc. If the token isn't a tokenID, this calls parseFunc.
+//
+// Ex. A function name is present `(module (func $math.pi (result f32))`
+//                        records math.pi here --^
+//                               parseFunc resumes here --^
+//
+// Ex. No function name `(module (func (result f32))`
+//              calls parseFunc here --^
+func (p *moduleParser) parseFuncName(tok tokenType, tokenBytes []byte, line, col uint32) error {
+	if tok == tokenID { // Ex. $main
+		na := &wasm.NameAssoc{Index: p.currentFuncIndex, Name: string(stripDollar(tokenBytes))}
+		p.module.funcNames = append(p.module.funcNames, na)
+		p.tokenParser = p.parseFunc
+		return nil
+	}
+	return p.parseFunc(tok, tokenBytes, line, col)
+}
+
+// parseFunc is the second parser inside the ed function field. This passes control to the typeParser until
```

Comment:
Perhaps is `ed` a typo? > `the ed function`

Reply:
yep! it was partial deletion of "imported function"